realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -147,6 +147,14 @@ public void onInit(OsSharedRealm sharedRealm) {
         this.shouldCloseSharedRealm = false;
     }
 
+    // FIXME Could we put this in Realm.getApplicationContext() only. Do not know why this lives
+    //  on BaseRealm as it is never initialized/used for DynamicRealm, so might as well just
+    //  be on Realm.
+    // FIXME Review public exposure - Needed from <X>Configuration.Builder. Can we expose this? Alternatively required as argument to <X>Configruation.Builders
+    public static Context getApplicationContext() {
```

Comment:
We mostly use this to check that people called `Realm.init()` as it loaded the native code and make sure we have a reference to `getFilesDir`.

If we must make it public, it would make most sense to have it on Realm I think.

Reply:
I added it as `Realm.getApplicationContext()`